### DISCUSSION TITLE
Why does it work with multiple-entity-row but not template-entity-row?

### DISCUSSION THREAD

**Discussion by gabrielvisser** [jcwillox/lovelace-paper-buttons-row#70] [Q&A] [answered]
Does anyone know why?

I can embed a button in a multiple-entity-row:
![image](https://user-images.githubusercontent.com/61377476/146786723-04301c71-6b55-46d4-b3f8-d4ff4f858728.png)

But not template-entity-row:
![image](https://user-images.githubusercontent.com/61377476/146786761-071b6110-b72f-448c-ba2b-3947f4347dab.png)

**Comment by jcwillox** [comment, marked as answer]
So it depends on how the row is constructed, the plugin looks for the `generic-entity-row` component that many rows use, rows that do not use that underlying component will not be detected. This is why you can't embed paper buttons in itself. It's necessary to do it this way as `generic-entity-row` has a consistent structure, meaning we can reliably extract the config and insert the buttons. Support could be added on a per-row basis, but this would be a lot of work to maintain and likely break over time.

Now in the case of `template-entity-row`, it does use `generic-entity-row` correctly and the buttons are actually inserted, however, it does something weird and hides the `generic-entity-row` and has a separate wrapper section that is visible.

![template-entity-row](https://user-images.githubusercontent.com/39120423/146928297-db683b87-2607-49f9-9257-3c92e6f142e5.png)

**Reply by gabrielvisser** [reply]
Clear explanation, thank you :)
